Automattic/sensei
# Make course list filter single block and implement student course filter

Implements https://github.com/Automattic/sensei/issues/5561

### Changes proposed in this Pull Request

- Made all the filters part of a single block where each of them can be toggled on and off.

- Added a student course state filter for Course List block. This is a course list block representation of our Student Courses block. All the options behave similarly to the filter header of that block. And it renders only when a student is logged in.
![image](https://user-images.githubusercontent.com/6820724/188232681-e4d7afd2-a8b8-4c05-adb6-5d69a417577a.png)

- Added the icon for filter

### Testing instructions

- Create some courses
- Create some course categories
- Assign some courses in some categories
- Add a Course List block on any page
- Add Course List Filter block inside the Course List block
- Try toggling on and off all the filters, make sure they are rendered in frontend and editor accordingly
- Make sure Student Course filter doesn't render if you are not logged in
- Try filtering by changing the dropdown value to different course categories, Featured options and Student Course state

### Screenshot / Video

https://user-images.githubusercontent.com/6820724/188775128-522ac744-085f-46db-aec4-439f9a68ea2f.mov
